pkgs/applications/networking/instant-messengers/matrix-commander/default.nix

```diff
@@ -1,38 +1,63 @@
-{ stdenv, lib, fetchFromGitHub, cacert, python3 }:
+{ lib
+, fetchFromGitHub
+, buildPythonApplication
+, cacert
+, setuptools
+, matrix-nio
+, python-magic
+, markdown
+, pillow
+, urllib3
+, aiofiles
+, notify2
+, dbus-python
+, xdg
+, python-olm
+}:
 
-stdenv.mkDerivation {
+buildPythonApplication rec {
   pname = "matrix-commander";
-  version = "unstable-2021-08-05";
+  version = "2.37.3";
 
   src = fetchFromGitHub {
     owner = "8go";
     repo = "matrix-commander";
-    rev = "7ab3fd9a0ef4eb19d882cb3701d2025b4d41b63a";
-    sha256 = "sha256-WWf7GbJxGlqIdsS1d0T1DO0WN2RBepHGgJrl/nt7UIg=";
+    rev = "v${version}";
+    sha256 = "sha256-X5tCPR0EqY1dxViwh8/tEjJM2oo81L3H703pPzWzUv8=";
   };
 
-  buildInputs = [
-    cacert
-    (python3.withPackages(ps: with ps; [
-      matrix-nio
-      magic
-      markdown
-      pillow
-      urllib3
-      aiofiles
-      notify2
-    ]))];
-
-  installPhase = ''
-    runHook preInstall
+  format = "pyproject";
 
-    mkdir -p $out/bin
-    cp $src/matrix-commander.py $out/bin/matrix-commander
-    chmod +x $out/bin/matrix-commander
+  postPatch = ''
+    # Dependencies already bundled with Python
+    sed -i \
+      -e '/uuid/d' \
+      -e '/argparse/d' \
+      -e '/asyncio/d' \
+      -e '/datetime/d' \
+      setup.cfg requirements.txt
 
-    runHook postInstall
+    # Dependencies not correctly detected
+    sed -i \
+      -e '/dbus-python/d' \
```

Comment:
Isn't this just dbus?

Reply:
It's listed as `dbus-python` in
https://github.com/8go/matrix-commander/blob/7cd410247d64a9b202e2f01c4375339b07ef80ab/requirements.txt#L6
and
https://github.com/8go/matrix-commander/blob/7cd410247d64a9b202e2f01c4375339b07ef80ab/setup.cfg#L37

I tried adding both `dbus` and `dbus-python` as build inputs, but this results in the following error (when not using `sed` to remove it from `requirements.txt`):

```
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./matrix_commander-2.37.3-py3-none-any.whl
Requirement already satisfied: python-magic in /nix/store/l19ka7yqf5cvb38vnb2h7ivh36lgwz85-python3.9-python-magic-0.4.26/lib/python3.9/site-packages (from matrix-commander==2.37.3) (0.4.26)
Requirement already satisfied: markdown in /nix/store/fkr0fvyd9vh0fddp0cxmlf0ziqji9pb1-python3.9-markdown-3.3.7/lib/python3.9/site-packages (from matrix-commander==2.37.3) (3.3.7)
ERROR: Could not find a version that satisfies the requirement dbus-python (from matrix-commander) (from versions: none)
ERROR: No matching distribution found for dbus-python
```